logstash-core/src/main/java/org/logstash/log/CustomLogEventSerializer.java

```diff
@@ -94,6 +96,16 @@ private void writeStructuredMessage(StructuredMessage message, JsonGenerator gen
         }
     }
 
+    private static String renameParamNameIfClashingWithMessage(Map.Entry<Object, Object> entry) {
```

Comment:
>```logger.info('this is test', :message => 'bar', :message_1' => 'foo')```

It's possibile but little bit unrealistic given that those logger invocations come from code the inside Logstash code base.

>```logger.info('this is test', hash_with_messages_keys)```

This could be a problem, where `hash_with_messages_keys` id something like
```json
{ "message": "blalba", "message_1": "blabla 1" }
```

and it's out of code control.

In general, whatever policy we apply to avoid collision, for example move the duplicate field in some nested map, it is subject to this latest problem you are exposing, because maps can overlap on root names, but also on nested structures.